src/views/notifications/CloseAllPositionsNotification.tsx

```diff
@@ -0,0 +1,114 @@
+import styled from 'styled-components';
+
+import { AbacusOrderStatus } from '@/constants/abacus';
+import { ButtonAction, ButtonSize } from '@/constants/buttons';
+import { STRING_KEYS } from '@/constants/localization';
+import { LocalCloseAllPositionsData } from '@/constants/trade';
+
+import { useStringGetter } from '@/hooks/useStringGetter';
+import { useSubaccount } from '@/hooks/useSubaccount';
+
+import { Button } from '@/components/Button';
+import { Details } from '@/components/Details';
+import { LoadingSpinner } from '@/components/Loading/LoadingSpinner';
+// eslint-disable-next-line import/no-cycle
+import { Notification, NotificationProps } from '@/components/Notification';
+
+import { OrderStatusIcon } from '../OrderStatusIcon';
+
+type ElementProps = {
+  localCloseAllPositions: LocalCloseAllPositionsData;
+};
+
+export const CloseAllPositionsNotification = ({
+  isToast,
+  localCloseAllPositions,
+  notification,
+}: NotificationProps & ElementProps) => {
+  const stringGetter = useStringGetter();
+  const { closeAllPositions } = useSubaccount();
+
+  const numPositions = localCloseAllPositions.submittedOrderClientIds.length;
+  const numClosed = localCloseAllPositions.filledOrderClientIds.length;
+  const numFailed = localCloseAllPositions.failedOrderClientIds.length;
+
+  const isPending = numClosed + numFailed < numPositions;
+  const allFilled = numClosed === numPositions;
+  const allCanceled = numFailed === numPositions;
+
+  let closePositionsStatus = <LoadingSpinner tw="text-color-accent [--spinner-width:0.9375rem]" />;
+  if (!isPending) {
+    let [statusStringKey, statusForIcon] = [
+      STRING_KEYS.PARTIALLY_FILLED,
+      AbacusOrderStatus.PartiallyCanceled.rawValue,
+    ];
+    if (allFilled) {
+      [statusStringKey, statusForIcon] = [
+        STRING_KEYS.ORDER_FILLED,
+        AbacusOrderStatus.Filled.rawValue,
+      ];
+    } else if (allCanceled) {
+      [statusStringKey, statusForIcon] = [STRING_KEYS.CANCELED, AbacusOrderStatus.Pending.rawValue];
+    }
+
+    closePositionsStatus = (
```

Comment:
super nit, I wonder if we could share a base notification with `CancelAllNotification` here 👀 (would we want try again on that one as well?)

Reply:
mmm let me think on this -- we'll also need to support limit close which might have extra logic on this one; i can have a followup pr on this if it makes sense!